src/platforms/Poap/ShowMintLinkButton.tsx

```diff
@@ -0,0 +1,92 @@
+import {
+  Box,
+  ButtonProps,
+  Center,
+  HStack,
+  Icon,
+  Link as LinkButton,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalHeader,
+  ModalOverlay,
+  Spinner,
+  Text,
+} from "@chakra-ui/react"
+import Button from "components/common/Button"
+import ErrorAlert from "components/common/ErrorAlert"
+import { Modal } from "components/common/Modal"
+import { ArrowSquareOut, CheckCircle } from "phosphor-react"
+import useClaimText from "platforms/SecretText/hooks/useClaimText"
+import { PropsWithChildren } from "react"
+import { Rest } from "types"
+import { useAccount } from "wagmi"
+
+export type ShowMintLinkButtonProps = { rolePlatformId: number } & ButtonProps & Rest
+
+export const ShowMintLinkButton: React.FC<
+  PropsWithChildren<ShowMintLinkButtonProps>
+> = ({ rolePlatformId, children, ...rest }) => {
+  const { address } = useAccount()
+  const {
+    isLoading,
+    error,
+    response,
+    modalProps: { isOpen, onOpen, onClose },
+  } = useClaimText(rolePlatformId)
+  const httpsLink = response?.uniqueValue?.replace("http://", "https://")
+
+  return (
+    <>
+      <Button onClick={onOpen} {...rest}>
+        {children}
+      </Button>
+      <Modal isOpen={isOpen} onClose={onClose}>
```

Comment:
Now this modal is duplicated, could you please refactor it so we use the same modal both in the claim flow and in the "show mint link" flow?